clippy_dev/src/new_lint.rs

```diff
@@ -38,25 +39,35 @@ pub fn create(
     pass: Option<&String>,
     lint_name: Option<&String>,
     category: Option<&String>,
+    ty: Option<&String>,
     msrv: bool,
 ) -> io::Result<()> {
     let lint = LintData {
-        pass: pass.expect("`pass` argument is validated by clap"),
+        pass: pass.map_or("", String::as_str),
         name: lint_name.expect("`name` argument is validated by clap"),
         category: category.expect("`category` argument is validated by clap"),
+        ty: ty.map(String::as_str),
         project_root: clippy_project_root(),
     };
 
     create_lint(&lint, msrv).context("Unable to create lint implementation")?;
     create_test(&lint).context("Unable to create a test for the new lint")?;
-    add_lint(&lint, msrv).context("Unable to add lint to clippy_lints/src/lib.rs")
+
+    if lint.ty.is_none() {
+        add_lint(&lint, msrv).context("Unable to add lint to clippy_lints/src/lib.rs")?;
+    }
+
+    Ok(())
 }
 
 fn create_lint(lint: &LintData<'_>, enable_msrv: bool) -> io::Result<()> {
```

Comment:
Just an optional idea. Maybe it would be nice, if this would print the generated file name to the console to let the user know where they should start implementing the lint. Something simple like: `println!("Generated lint file: {}")`

Reply:
Good call :slightly_smiling_face:.